### ISSUE TITLE
PHP 8 reports deprecation warnings due to optional parameter ordering

### ISSUE THREAD

**Issue by mszabo-wikia** [SemanticMediaWiki/SemanticMediaWiki#4701]
### Setup and configuration

- SMW version: 3.2.0-alpha, git: 3f86d00
- MW version: 1.35.0-alpha, git: 66f7b12
- PHP version: PHP 8.0.0-dev
- DB system (MySQL, Blazegraph, etc.) and version: MySQL 5.7.29

### Issue
PHP 8 is stricter about functions that have required parameters after optional ones, and now raises a deprecation warning for such functions. This is causing SMW to log deprecation warnings when run on PHP 8.

While SMW doesn't yet officially support PHP 8, I'm creating this issue in the hope that it will assist with eventual preparations for supporting PHP 8.

See also the upstream MediaWiki issue for tracking this PHP change.[1]

---
[1] https://phabricator.wikimedia.org/T248078

### Stack trace
Here is a non-exhaustive list of deprecation warnings logged by a test run on PHP 8:
```
Deprecated: Required parameter $fname follows optional parameter $conditions in /mediawiki/extensions/SemanticMediaWiki/src/MediaWiki/Connection/Database.php on line 264

Deprecated: Required parameter $vars follows optional parameter $args in /mediawiki/extensions/SemanticMediaWiki/src/SetupFile.php on line 458

Deprecated: Required parameter $key follows optional parameter $subject in /mediawiki/extensions/SemanticMediaWiki/src/EntityCache.php on line 225

Deprecated: Required parameter $expected follows optional parameter $caption in /mediawiki/extensions/SemanticMediaWiki/tests/phpunit/includes/DataValues/UriValueTest.php on line 31
```

### Steps to reproduce
Run the SMW test suite using PHP 8.

**Comment by JeroenDeDauw** [comment]
Oh boy... there is a lot of code with optional parameters before required ones. Did that really get deprecated >_>

**Comment by mwjames** [comment]
@mszabo-wikia Much appreciated your effort here, especially providing a PR to address the issue.

**Comment by kghbln** [comment]
From the comments here and on the pull request I conclude that everything was done to fluff the code for PHP8 concerning the deprecation warnings reported with this issue. Thus closing. The changes will be shipped with SMW 3.2.0 I hope to get out to the world in May or June 2020. Keeping fingers crossed.
